velox/serializers/tests/PrestoSerializerTest.cpp

```diff
@@ -320,7 +332,8 @@ class PrestoSerializerTest
     auto deserialized = deserialize(rowType, serialized, serdeOptions);
 
     assertEqualVectors(original, deserialized);
-    assertEqualEncoding(original, deserialized);
+    // Dictionaries may get flattened depending on the nature of the data.
+    assertEqualEncoding(original, deserialized, true);
```

Comment:
should we pass allowFlatteningDictionaries here instead of always true and then explicitly set in in tests that require it? as setting this to always true might give us false positives for the test cases that expect encodings to always be preserved